# 01. 필터링, 최솟값, 중앙값

1. 'f5' 컬럼이 0이 아닌 데이터(행)를 구하시오.
2. 앞에서 구한 데이터에 'Views' 컬럼 결측치를 'views' 컬럼의 최솟값으로 채워주세요.
3. 그리고 'views' 컬럼의 중앙값을 계산해 정수로 구하시오.

In [21]:
import pandas as pd 
df = pd.read_csv("../bigdata_analyst_cert/part1/ch3/type1_data1.csv")

In [22]:
df.head()

id   age city    f1  f2    f3    f4         f5  subscribed   views
0  id01   2.0   서울   NaN   0  gold  ENFJ  91.297791  2024-07-16  6820.0
1  id02   9.0   서울  70.0   1   NaN  ENFJ  60.339826  2024-05-12  2534.0
2  id03  27.0   서울  61.0   1  gold  ISTJ  17.252986  2024-03-16  7312.0
3  id04  75.0   서울   NaN   2   NaN  INFP  52.667078  2024-07-21   493.0
4  id05  24.0   서울  85.0   2   NaN  ISFJ  29.269869  2024-03-07  1338.0

In [23]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120 entries, 0 to 119
Data columns (total 10 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   id          120 non-null    object 
 1   age         120 non-null    float64
 2   city        120 non-null    object 
 3   f1          89 non-null     float64
 4   f2          120 non-null    int64  
 5   f3          91 non-null     object 
 6   f4          120 non-null    object 
 7   f5          120 non-null    float64
 8   subscribed  120 non-null    object 
 9   views       116 non-null    float64
dtypes: float64(4), int64(1), object(5)
memory usage: 9.5+ KB


In [24]:
# df['f5'] 가 0인 행(rows)은 18개, 0이 아닌 행은 102개
print(df[df['f5'] == 0].shape)
print(df[df['f5'] != 0].shape)

(18, 10)
(102, 10)


In [25]:
df_f5 = df[df['f5'] != 0]
df_f5.shape

(102, 10)

In [26]:
# 앞의 데이터에 views 컬럼의 결측값 (4개 행)을 views 컬럼의 최솟값(42.0)으로 대체하기
print(df_f5['views'].isna().sum())
print(df_f5['views'].min())

4
42.0


In [27]:
df_f5.describe()

age          f1          f2          f5         views
count  102.000000   71.000000  102.000000  102.000000     98.000000
mean    50.767647   65.563380    0.656863   56.614206   6071.632653
std     30.183600   19.394942    0.724331   28.288891   4247.580450
min    -13.500000   12.000000    0.000000    9.796378     42.000000
25%     27.750000   50.500000    0.000000   30.279652   3103.750000
50%     52.000000   62.000000    1.000000   64.113099   5997.000000
75%     76.500000   78.000000    1.000000   80.138280   8432.500000
max    100.000000  111.000000    2.000000   98.429899  21550.000000

In [28]:
df_f5[df_f5['views'].isna()]

id   age city    f1  f2   f3    f4         f5  subscribed  views
20  id20  11.0   서울  51.0   1  NaN  INTJ  91.297791  2024-07-16    NaN
39  id39  56.0   부산  50.0   0  NaN  INFJ  33.308999  2024-12-22    NaN
48  id48  18.0   대구   NaN   0  NaN  ENFP  20.129444  2024-03-25    NaN
82  id82  48.0   경기   NaN   0  NaN  ENTJ  37.113739  2024-10-17    NaN

In [29]:
views_na_index = df_f5[df_f5['views'].isna()].index

In [30]:
df_f5['views'] = df_f5['views'].fillna(df_f5['views'].min())

C:\Users\ckwjd\AppData\Local\Temp\ipykernel_16972\2644754277.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_f5['views'] = df_f5['views'].fillna(df_f5['views'].min())


In [31]:
df_f5['views'].isna().sum()

np.int64(0)

In [32]:
df_f5.iloc[views_na_index,:]

id   age city    f1  f2   f3    f4         f5  subscribed  views
20  id20  11.0   서울  51.0   1  NaN  INTJ  91.297791  2024-07-16   42.0
39  id39  56.0   부산  50.0   0  NaN  INFJ  33.308999  2024-12-22   42.0
48  id48  18.0   대구   NaN   0  NaN  ENFP  20.129444  2024-03-25   42.0
82  id82  48.0   경기   NaN   0  NaN  ENTJ  37.113739  2024-10-17   42.0

In [33]:
# views 컬럼의 중앙값을 정수로 구하기

df_f5['views'].median()

np.float64(5924.0)

답: 5924!

## 02. 카테고리, 인덱스, 문자열 슬라이싱

1. 'subscribed' 컬럼에서 가장 빈도수가 많은 날짜를 구하시오.
2. 앞에서 구한 날짜의 일(day) 값을 정수로 구하시오.

In [34]:
import pandas as pd 
df = pd.read_csv("../bigdata_analyst_cert/part1/ch3/type1_data1.csv")

In [35]:
df.head()

id   age city    f1  f2    f3    f4         f5  subscribed   views
0  id01   2.0   서울   NaN   0  gold  ENFJ  91.297791  2024-07-16  6820.0
1  id02   9.0   서울  70.0   1   NaN  ENFJ  60.339826  2024-05-12  2534.0
2  id03  27.0   서울  61.0   1  gold  ISTJ  17.252986  2024-03-16  7312.0
3  id04  75.0   서울   NaN   2   NaN  INFP  52.667078  2024-07-21   493.0
4  id05  24.0   서울  85.0   2   NaN  ISFJ  29.269869  2024-03-07  1338.0

In [36]:
df['subscribed'].describe()

count            120
unique            96
top       2025-02-17
freq              16
Name: subscribed, dtype: object

In [ ]:
# subscribed 컬럼에서 가장 빈도수가 많은 날짜 = 2025-02-17
df['subscribed'].value_counts()

subscribed
2025-02-17    16
2024-06-21     3
2024-07-16     2
2024-04-03     2
2024-02-09     2
              ..
2024-05-06     1
2024-10-02     1
2024-12-02     1
2025-01-29     1
2025-01-30     1
Name: count, Length: 96, dtype: int64

In [44]:
sub_mode = df['subscribed'].value_counts().index[0]
print(sub_mode)

2025-02-17


In [45]:
print(type(sub_mode))

<class 'str'>


답은 17!

| 추가로,,, subscribed 컬럼을 날짜형으로 변경한 다음에 일만 발라내는 작업을... 해보면 좋을 듯?

## 03. 파생변수, 정렬, 인덱싱

1. 결측치가 있는 데이터(행)을 제거하시오.
2. 'Views' 컬럼을 'f1' 컬럼으로 나눈 값을 새로운 컬럼으로 추가하시오.
3. 새로운 컬럼 값 중 가장 큰 값을 가진 행의 age를 정수로 구하시오.

In [53]:
import pandas as pd 
df = pd.read_csv("../bigdata_analyst_cert/part1/ch3/type1_data1.csv")

In [54]:
# f1 컬럼 - 결측치 없는 행 89개
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120 entries, 0 to 119
Data columns (total 10 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   id          120 non-null    object 
 1   age         120 non-null    float64
 2   city        120 non-null    object 
 3   f1          89 non-null     float64
 4   f2          120 non-null    int64  
 5   f3          91 non-null     object 
 6   f4          120 non-null    object 
 7   f5          120 non-null    float64
 8   subscribed  120 non-null    object 
 9   views       116 non-null    float64
dtypes: float64(4), int64(1), object(5)
memory usage: 9.5+ KB


In [55]:
df.isna().sum()

id             0
age            0
city           0
f1            31
f2             0
f3            29
f4             0
f5             0
subscribed     0
views          4
dtype: int64

In [56]:
df_drop_na = df.dropna()

In [57]:
df_drop_na.info()

<class 'pandas.core.frame.DataFrame'>
Index: 70 entries, 2 to 119
Data columns (total 10 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   id          70 non-null     object 
 1   age         70 non-null     float64
 2   city        70 non-null     object 
 3   f1          70 non-null     float64
 4   f2          70 non-null     int64  
 5   f3          70 non-null     object 
 6   f4          70 non-null     object 
 7   f5          70 non-null     float64
 8   subscribed  70 non-null     object 
 9   views       70 non-null     float64
dtypes: float64(4), int64(1), object(5)
memory usage: 6.0+ KB


In [58]:
df_drop_na.isna().sum()

id            0
age           0
city          0
f1            0
f2            0
f3            0
f4            0
f5            0
subscribed    0
views         0
dtype: int64

In [59]:
# views 컬럼을 f1 컬럼의 값으로 나눈 값을 새로운 컬럼으로 추가
df_drop_na['new'] = df_drop_na['views'] / df_drop_na['f1']
df_drop_na

C:\Users\ckwjd\AppData\Local\Temp\ipykernel_16972\2030598373.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_drop_na['new'] = df_drop_na['views'] / df_drop_na['f1']


id   age city     f1  f2      f3    f4         f5  subscribed  \
2     id03  27.0   서울   61.0   1    gold  ISTJ  17.252986  2024-03-16   
5     id06  22.0   서울   57.0   0     vip  INTP  20.129444  2024-09-12   
7     id08  38.0   서울  101.0   1  silver  INFJ  83.685380  2024-03-06   
9     id10  95.0   서울   74.0   1    gold  ISFP  98.429899  2024-04-03   
10   id100  47.0   경기   53.0   0     vip  ESFP  33.308999  2024-02-21   
..     ...   ...  ...    ...  ..     ...   ...        ...         ...   
115  id114  22.0   대구   23.0   0    gold  INTP   0.000000  2025-02-17   
116  id115  23.0   부산   65.0   0     vip  ENTP   0.000000  2025-02-17   
117  id116  23.0   서울   12.0   1  silver  INFP   0.000000  2025-02-17   
118  id117  23.0   대구   65.0   2    gold  INFP   0.000000  2025-02-17   
119  id118  24.0   부산   94.0   1     vip  ESFJ   0.000000  2025-02-17   

       views         new  
2     7312.0  119.868852  
5    21550.0  378.070175  
7     3260.0   32.277228  
9     9992.0  135.027027  
10   15535.0  293.113208  
..       ...         ...  
115   9747.0  423.782609  
116   5628.0   86.584615  
117   1267.0  105.583333  
118   6543.0  100.661538  
119   2356.0   25.063830  

[70 rows x 11 columns]

In [63]:
max_value = df_drop_na['new'].max()
df_drop_na[df_drop_na['new'] == max_value]

id   age city    f1  f2    f3    f4   f5  subscribed   views  \
115  id114  22.0   대구  23.0   0  gold  INTP  0.0  2025-02-17  9747.0   

            new  
115  423.782609

In [64]:
df_drop_na[df_drop_na['new'] == max_value]['age']

115    22.0
Name: age, dtype: float64

답은 22~

## 04. 값 변경, 정렬, 합계

1. 'Views' 컬럼의 결측 데이터를 0으로 대체하시오.
2. 'Views' 컬럼에서 상위 10번째 값을 구하시오.
3. 'Views' 컬럼에서 상위 10개의 값을 상위 10번째 값으로 대체하시오.
4. 'views' 컬럼의 전체 합을 정수로 구하시오.

In [65]:
import pandas as pd 
df = pd.read_csv("../bigdata_analyst_cert/part1/ch3/type1_data1.csv")

In [66]:
df['views'].isna().sum()

np.int64(4)

In [67]:
na_index = df[df['views'].isna()].index
na_index

Index([20, 39, 48, 82], dtype='int64')

In [70]:
df.loc[na_index]

id   age city    f1  f2   f3    f4         f5  subscribed  views
20  id20  11.0   서울  51.0   1  NaN  INTJ  91.297791  2024-07-16    NaN
39  id39  56.0   부산  50.0   0  NaN  INFJ  33.308999  2024-12-22    NaN
48  id48  18.0   대구   NaN   0  NaN  ENFP  20.129444  2024-03-25    NaN
82  id82  48.0   경기   NaN   0  NaN  ENTJ  37.113739  2024-10-17    NaN

In [ ]:
# view 컬럼의 결측 데이터를 0으로 치환
df['views'] = df['views'].fillna(0)

df.loc[na_index]

id   age city    f1  f2   f3    f4         f5  subscribed  views
20  id20  11.0   서울  51.0   1  NaN  INTJ  91.297791  2024-07-16    0.0
39  id39  56.0   부산  50.0   0  NaN  INFJ  33.308999  2024-12-22    0.0
48  id48  18.0   대구   NaN   0  NaN  ENFP  20.129444  2024-03-25    0.0
82  id82  48.0   경기   NaN   0  NaN  ENTJ  37.113739  2024-10-17    0.0

In [79]:
df['views'].sort_values(ascending=False).head(10)

5      21550.0
56     19589.0
83     19139.0
32     17421.0
10     15535.0
100    15535.0
104    13543.0
114    10346.0
9       9992.0
113     9877.0
Name: views, dtype: float64

In [76]:
# views 컬럼에서 상위 10번째 값

top_10 = df['views'].sort_values(ascending=False).head(10).iloc[9]
top_10

np.float64(9877.0)

In [78]:
# views 컬럼에서 상위 10개의 값을 상위 10번째 값으로 대체

def exchange(x, y):
    if x > y:
        x = y

df['new_views'] = df['views'].apply(lambda x: top_10 if x > top_10 else x)
df['new_views'].sort_values(ascending=False).head(10)

5      9877.0
9      9877.0
10     9877.0
32     9877.0
100    9877.0
114    9877.0
83     9877.0
56     9877.0
113    9877.0
104    9877.0
Name: new_views, dtype: float64

In [80]:
df['new_views'].sum()

np.float64(652812.0)

답은 652812...?

## 05. 문자열 슬라이싱, 파생변수, 평균값

1. 'f4' 컬럼 데이터에 'FJ'가 포함된 데이터를 찾으시오.
2. 찾은 데이터 중에서 'f2' 컬럼의 평균값을 구하시오. (반올림 후 소수 둘째 자리까지 계산)

In [81]:
import pandas as pd 
df = pd.read_csv("../bigdata_analyst_cert/part1/ch3/type1_data1.csv")

In [85]:
df_fj = df[df['f4'].str.contains('FJ')]

In [86]:
df_fj['f2'].mean()

np.float64(0.6071428571428571)

답은 : 0.61

In [82]:
# f4 컬럼 데이터에 'FJ'가 포함된 데이터
help(df['f4'].str)

Help on StringMethods in module pandas.core.strings.accessor object:

class StringMethods(pandas.core.base.NoNewAttributesMixin)
 |  StringMethods(data) -> 'None'
 |  
 |  Vectorized string functions for Series and Index.
 |  
 |  NAs stay NA unless handled otherwise by a particular method.
 |  Patterned after Python's string methods, with some inspiration from
 |  R's stringr package.
 |  
 |  Examples
 |  --------
 |  >>> s = pd.Series(["A_Str_Series"])
 |  >>> s
 |  0    A_Str_Series
 |  dtype: object
 |  
 |  >>> s.str.split("_")
 |  0    [A, Str, Series]
 |  dtype: object
 |  
 |  >>> s.str.replace("_", "")
 |  0    AStrSeries
 |  dtype: object
 |  
 |  Method resolution order:
 |      StringMethods
 |      pandas.core.base.NoNewAttributesMixin
 |      builtins.object
 |  
 |  Methods defined here:
 |  
 |  __getitem__(self, key)
 |  
 |  __init__(self, data) -> 'None'
 |      Initialize self.  See help(type(self)) for accurate signature.
 |  
 |  __iter__(self) -> 'Iterator'
 |  

## 06. 필터링, 분산

1. 'f3' 컬럼이 gold이면서 'f2' 컬럼이 2인 데이터를 찾으시오.
2. 찾은 데이터에서 'f1' 컬럼의 분산을 구하시오. (반올림 후 소수 둘째 자리까지 계산)

## 07. 값 변경(연산), 필터링 절댓값

1. 모든 나이(age)에 1을 더하시오.
2. 20대의 'views' 평균과 30대의 'views' 평균의 절댓값 차이를 구하시오. (반올림 후 소수 둘째 자리까지 계산)

## 08. 시계열 데이터, 필터링, 데이터 개수

1. 'subscribed' 컬럼이 2024sus 2월인 데이터를 찾으시오.
2. 위에서 찾은 데이터 중 'f3' 컬럼이 gold인 데이터의 개수를 구하시오.

## 09. 필터링, 카테고리, 최빈값

1. 'views' 컬럼 값이 1000 이하인 데이터(결측치 제외)를 찾으시오.
2. 앞에서 구한 데이터 중 'f4' 컬럼의 최빈값을 구하시오.

## 10. 그룹핑, 최댓값, 정렬

1. 결측치가 있는 행을 삭제하시오.
2. 결측치가 삭제된 데이터를 사용하여 지역별(city) 평균을 계산하시오.
3. 앞에서 계산한 지역별 평균 데이터에서 'f2' 컬럼 값이 가장 큰 지역을 구하시오.

## 11. 슬라이싱, 사분위수, 결측치 제거

1. 데이터에서 결측치가 있는 데이터(행)를 모두 제거하시오.
2. 결측치가 제거된 데이터를 사용하여 앞에서부터 70% 데이터를 구하시오. (단, 데이터 70% 지점의 index가 소수점으로 계산될 경우 소수점 이하는 버림)
3. 앞에서 구한 70% 데이터 중 'views' 컬럼의 3사분위수에서 1사분위수를 뺀 값을 정수로 구하시오.

## 12. 결측치 처리, 최빈값, 데이터 개수

1. 결측치가 가장 많은 두 컬럼을 찾으시오.
2. 첫번째로 결측치가 많은 컬럼에서 결측치가 있는 데이터(행)를 삭제하시오.
3. 두 번째로 결측치가 많은 컬럼을 최빈값으로 대체하시오.
4. 'f3' 컬럼의 'gold' 값을 가진 데이터의 수를 정수형으로 구하시오.

## 13. 결측 데이터 찾기, 필터링, 평균값

1. 'f1' 컬럼에 결측치가 있는 데이터만 선택하시오.
2. 선택된 데이터에서 'age' 컬럼의 평균값을 구하시오. (반올림 후 소수 첫째 자리까지 계산)

## 14. 중복 데이터 제거, 값 변경, 데이터 개수

1. 중복 데이터를 제거하시오.
2. 'f3' 컬럼의 결측치는 0, 'silver'는 1, 'gold'는 2, 'vip'는 3으로 변환하시오.
3. 변환된 'f3' 컬럼의 총합을 정수형으로 구하시오.

## 15. 컬럼 삭제, 행 단위 합계, 필터링

1. 주어진 데이터에서 문자 자료형 컬럼을 삭제하시오.
2. 숫자 자료형 컬럼의 결측치를 0으로 대체하시오.
3. 각 행의 합이 3000보다 큰 값의 개수를 정수로 구하시오. (각 행의 합: 'age' + 'f1' + 'f2' + 'f5' + 'views')

## 16. 이상치, IQR

1. 'views' 컬럼의 1사분위수, 3사분위수 그리고 IQR를 계산하시오.
2. 이상치 조건에 맞는 데이터를 찾으시오. (이상치는 1사분위수 - (IQR * 1.5)보다 작은 값과 3사분위수 + (IQR * 1.5)보다 큰 값)
3. 이상치 데이터의 'views' 컬럼 합을 정수로 구하시오.

## 17. 이상치, 소수점 있는 데이터 찾기, 표준 편차

1. 'views' 컬럼의 표준편차를 구하시오.
2. 'age' 컬럼의 이상치(소수점 나이, 음수 나이, 0살)을 제거하고, 'views' 컬럼의 표준편차를 구하시오.
3. 이상치 제거 전후의 'views' 컬럼의 표준편차를 더하여, 반올림 후 소수 둘째 자리까지 구하시오.

## 18. 데이터(행) 기준 평균값, 인덱싱

1. index '2001' 데이터(행)에서 평균보다 큰 값의 개수를 구하시오.
2. index '2003' 데이터(행)에서 평균보다 작은 값의 개수를 구하시오.
3. 두 개수를 더하시오.

## 19. 결측치(뒤의 값으로 대체), 그룹합

1. 결측치를 바로 뒤에 있는 값으로 대체하시오. (바로 두의 값도 결측치일 경우, 뒤에 있는 데이터 중 가장 가까운 값으로 대체)
2. 'city'와 'f2' 컬럼을 기준으로 그룹합을 계산하시오.
3. 'views' 값이 세 번째로 큰 city 이름을 구하시오.

## 20. 시계열 데이터, 월별 집계, 인덱스

1. 연도 구분 없이 월별로 숫자형 컬럼의 합을 구하시오.
2. 합계 중 'views'가 가장 작은 값을 가진 월을 정수로 구하시오.

## 21. 시간 간의 차이 계산(분), 필터링

1. 예상 도착 시간보다 늦게 도착한 건수를 구하시오.
2. 이 중 거리가 7km 이상인 데이터의 수를 정수로 구하시오.

In [ ]:
import pandas as pd 
df = pd.read_csv("../bigdata_analyst_cert/part1/ch3/delivery_time.csv")

## 22. 시간 간의 차이 계산(분), 그룹핑

1. 주문 시간과 실제 도착 시간의 차이를 분 단위로 계산하시오.
2. 앱 종류별로 평균 도착 시간(분)을 계산하시오.
3. 평균적으로 가장 빠른 앱 종류를 찾고, 해당 앱의 평균 도착 시간을 분으로 반올림하여 정수로 구하시오.

In [ ]:
import pandas as pd 
df = pd.read_csv("../bigdata_analyst_cert/part1/ch3/delivery_time.csv")

## 23. 시간 간의 차이 계산(분), 비율

1. 각 결제 종류별로 실제 도착 시간이 예상 도착 시간보다 늦은 주문의 비율을 계산하시오.
2. 비율 중 가장 큰 값을 반올림하여 소수 둘째 자리까지 구하시오.

## 24. 그룹핑, 값 찾기, 필터링

1. 사용자별 주문 거리의 합계가 50km 이상인 사람들의 결제 방식을 구하시오.
2. 이 결제 방식 중 가장 빈도가 높은 수를 구하시오.

In [ ]:
import pandas as pd 
df = pd.read_csv("../bigdata_analyst_cert/part1/ch3/delivery_time.csv")

## 25. 시간 간의 차이 계산(일)

1. 각 사용자별로 첫 주문과 마지막 주문 사이의 시간 간격을 일 단위로 계산하시오.
2. 시간 간격이 1일 이하인 사용자를 제외하고, 나머지 사용자들의 평균 시간 간격(일 단위)을 계산하시오.
3. 평균 시간 간격보다 긴 시간 간격을 가진 사용자의 수를 정수로 구하시오.

## 26. 날짜와 시간 정보 변환, 비율

1. 주문이 가장 많이 발생한 연-월을 찾으시오.
2. 해당 연-월에 '배고팡' 앱을 통한 주문 중 '앱결제'로 결제된 주문의 비율을 계산하시오. (반올림 후 소수 둘째 자리까지 계산)

## 27. 시간 범위, 속도(km/h)

1. 점심시간(10시부터 13시 전까지)에 주문된 배달 데이터를 찾으시오.
2. 점심시간 주문 건 중 과속(평균 속도가 50km/h 이상)하는 주문 수를 정수로 구하시오.
* 배달시간 = 실제도착시간 - 주문시간
* 속도(km/h) = 거리(km)/시간(h)

## 28. 날짜와 시간, 문자열

1. 연도와 월을 기준으로 주문 수를 집계하시오.
2. 가장 많은 주문이 있었던 연도와 월을 예시와 같은 형식으로 숫자로만 구하시오. (예: 2024년 2월인 경우 '202402', 2024년 10월인 경우 '202410')

## 29. 함수, 월별 집계

1. 배달료 계산 기준표에 따라 각 주문에 대한 배달료를 계산하시오.
2. 월별로 대발료의 총합을 집계하시오.
3. 배달료가 가장 많이 발생한 월을 찾고, 그 월의 총 배달료를 정수로 구하시오.

[배달료 기준표]
* 5km 미만: 2,000원
* 5km 이상 ~ 10km 미만: 4,000원
* 10km 이상 ~ 15km 미만: 6,000원
* 15km 이상 ~ 20km 미만: 8,000원

## 30. 주말, 평일 구분

1. 주말 주문 건수와 평일 주문 건수를 구하시오.
2. 주말 주문 건수와 평일 주문 건수의 차이를 절대값으로 구하고 정수형으로 구하시오.

## 31. 문자열, 형 변환

1. 'user' 컬럼에서 user 뒤에 있는 숫자 값만 추출하시오.
2. 추출된 숫자 값을 모두 합한 값을 정수로 구하시오.

## 32. 합계(열 방향), 상위 값 선택

1. 수학, 영어, 국어 점수의 합을 구하시오.
2. 합이 가장 큰 상위 10명을 찾으시오.
3. 찾은 10명의 수학 평균 점수를 구하시오. (반ㅇ올림 후 정수 출력)

In [ ]:
import pandas as pd 
df = pd.read_csv("../bigdata_analyst_cert/part1/ch3/school_data.csv")

## 33. 데이터프레임 재구조화

1. 과목에 상관없이 점수가 가장 작은 점수 25개를 찾으시오.
2. 찾은 점수 25개의 합을 정수로 구하시오.

In [ ]:
import pandas as pd 
df = pd.read_csv("../bigdata_analyst_cert/part1/ch3/school_data.csv")

## 34. 데이터 합치기(concat)

1. school_data.csv와 school_data_science.csv의 학생 순서는 동일하다.
2. 학생별로 수학, 영어, 국어, 과학 점수의 평균을 구하시오.
3. 평균 점수가 60점 이상인 인원수를 계산하시오.

In [ ]:
import pandas as pd 
df = pd.read_csv("../bigdata_analyst_cert/part1/ch3/school_data.csv")
df_science = pd.read_csv("../bigdata_analyst_cert/part1/ch3/school_data_science.csv")

## 35. 데이터 합치기(merge)

1. school_data.csv와 school_data_social.csv 파일을 '이름'을 기준으로 합치시오.
2. 영어 교사가 장선생이면서 사회교사가 오선생인 학생들을 필터링하시오.
3. 필터링된 학생들의 수학 점수를 모두 더한 후 정수로 구하시오.

In [ ]:
import pandas as pd 
df = pd.read_csv("../bigdata_analyst_cert/part1/ch3/school_data.csv")
df_social = pd.read_csv("../bigdata_analyst_cert/part1/ch3/school_data_social.csv")